RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -221,214 +293,542 @@ double AntiElectronIDMVA6<TauType, ElectronType>::MVAValue(const TauVars& tauVar
+      var_wGwGSF_Endcap_[4] = elecVars.gsfTracklnPt;
+      var_wGwGSF_Endcap_[5] = elecNumHitsDiffOverSum;
+      var_wGwGSF_Endcap_[6] = std::log(std::max(0.01f, elecVars.chi2NormKF));
+      var_wGwGSF_Endcap_[7] = std::min(elecDeltaPinPoutOverPin, 1.f);
+      var_wGwGSF_Endcap_[8] = std::min(elecEecalOverPout, 20.f);
+      var_wGwGSF_Endcap_[9] = elecVars.deltaEta;
+      var_wGwGSF_Endcap_[10] = elecVars.deltaPhi;
+      var_wGwGSF_Endcap_[11] = std::min(elecVars.mvaInSigmaEtaEta, 0.01f);
+      var_wGwGSF_Endcap_[12] = std::min(elecVars.mvaInHadEnergy, 20.f);
+      var_wGwGSF_Endcap_[13] = std::min(elecVars.mvaInDeltaEta, 0.1f);
+      var_wGwGSF_Endcap_[14] = tauVars.etaAtEcalEntrance;
+      var_wGwGSF_Endcap_[15] = tauVars.leadChargedPFCandEtaAtEcalEntrance;
+      var_wGwGSF_Endcap_[16] = std::min(2.f, tauVars.leadChargedPFCandPt / std::max(1.f, tauVars.pt));
+      var_wGwGSF_Endcap_[17] = std::log(std::max(1.f, tauVars.pt));
+      var_wGwGSF_Endcap_[18] = tauVars.emFraction;
+      var_wGwGSF_Endcap_[19] = tauGammaMoms.signalPFGammaCandsIn;
+      var_wGwGSF_Endcap_[20] = tauGammaMoms.signalPFGammaCandsOut;
+      var_wGwGSF_Endcap_[21] = tauVars.leadPFChargedHadrHoP;
+      var_wGwGSF_Endcap_[22] = tauVars.leadPFChargedHadrEoP;
+      var_wGwGSF_Endcap_[23] = tauVars.visMassIn;
+      var_wGwGSF_Endcap_[24] = tauGammaMoms.gammaEtaMomIn;
+      var_wGwGSF_Endcap_[25] = tauGammaMoms.gammaEtaMomOut;
+      var_wGwGSF_Endcap_[26] = tauGammaMoms.gammaPhiMomIn;
+      var_wGwGSF_Endcap_[27] = tauGammaMoms.gammaPhiMomOut;
+      var_wGwGSF_Endcap_[28] = tauGammaMoms.gammaEnFracIn;
+      var_wGwGSF_Endcap_[29] = tauGammaMoms.gammaEnFracOut;
+      var_wGwGSF_Endcap_[30] = tauVars.dCrackEta;
+      mvaValue = mva_wGwGSF_EC_->GetClassifier(var_wGwGSF_Endcap_.data());
     }
   }
   return mvaValue;
 }
+////
+template <class TauType, class ElectronType>
+double AntiElectronIDMVA6<TauType, ElectronType>::mvaValuePhase2(const TauVars& tauVars,
+                                                                 const TauGammaMoms& tauGammaMoms,
+                                                                 const ElecVars& elecVars) {
+  if (!isInitialized_) {
+    throw cms::Exception("ClassNotInitialized") << " AntiElectronMVA6 not properly initialized !!\n";
+  }
+
+  double mvaValue = -99.;
 
+  //do not consider tau candidates outside the HGCal border at |eta|=3
+  if (std::abs(tauVars.etaAtEcalEntrance) > 3.0) {
+    return mvaValue;
+  }
+
+  float elecDeltaPinPoutOverPin = (elecVars.pIn > 0.0) ? (std::abs(elecVars.pIn - elecVars.pOut) / elecVars.pIn) : 1.0;
+  float elecEecalOverPout = (elecVars.pOut > 0.0) ? (elecVars.eEcal / elecVars.pOut) : 20.0;
+  float elecNumHitsDiffOverSum =
+      ((elecVars.gsfNumHits + elecVars.kfNumHits) > 0.0)
+          ? ((elecVars.gsfNumHits - elecVars.kfNumHits) / (elecVars.gsfNumHits + elecVars.kfNumHits))
+          : 1.0;
+
+  if (tauGammaMoms.signalPFGammaCandsIn == 0 && tauVars.hasGsf < 0.5) {
```

Comment:
the > 0.5 condition for `hasGsf` is used multiple times; i suggest making this a named constant (to enforce consistency)

Reply:
0.5 is just a number between 0 and 1 used check a boolean coded as float number and avoid comparison with 0. or 1. The number is float as it is required by TMVA (or historically was). It is not very nice, but I do not think it is error prone, so I would like to stay with current version (used also in non-phase-2 code).

Reply:
My suggestion is just to have `constexpr float gsfCut = 0.5` and then use `gsfCut` everywhere to ensure consistency and clarity. But if the current usage predates this PR, it's not strictly necessary.